src/api/ArtifactoryClient.ts

```diff
@@ -0,0 +1,123 @@
+import * as stream from 'stream';
```

Comment:
It is better to stop separating public and non-public classes/types/interfaces by means of repository structure. In Node.js ecosystem we, fortunately, have reexports and entry points, and we could use them to expose all the API.

Adding or removing something from API should not trigger moving files back and forth in the repo, like sometimes happens with public includes in C++ projects.

Btw, the module has inconsistent file naming.